cmd/manager/main.go

```diff
@@ -228,7 +239,6 @@ func newRootCommand() *cobra.Command {
 			// listen for interrupts or the Linux SIGTERM signal and cancel
 			// our context, which the leader election code will observe and
 			// step down
-			ch := make(chan os.Signal, 1)
 			signal.Notify(ch, os.Interrupt, syscall.SIGTERM)
```

Comment:
You don't want to do this part any more for the leader-election context. The TERM signal should only directly causing the controllers to stop and not also directly causing the release of the leader-election lock.